jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -216,6 +216,111 @@ ecma_new_ecma_string_from_utf8 (const lit_utf8_byte_t *string_p, /**< utf-8 stri
 } /* ecma_new_ecma_string_from_utf8 */
 
 /**
+ * Allocate new ecma-string and fill it with characters from the utf8 string
+ * It converts all 4-bytes long unicode sequence to two 3-bytes long sequence
```

Comment:
Allocate a new ecma-string and initialize it from the utf8 string argument.
All 4-bytes long unicode sequences are converted into two 3-bytes long sequences.